pyo3-build-config/src/impl_.rs

```diff
@@ -1078,33 +1188,56 @@ fn search_lib_dir(path: impl AsRef<Path>, cross: &CrossCompileConfig) -> Vec<Pat
 ///
 /// [1]: https://github.com/python/cpython/blob/3.8/Lib/sysconfig.py#L348
 fn cross_compile_from_sysconfigdata(
-    cross_compile_config: CrossCompileConfig,
-) -> Result<InterpreterConfig> {
-    let sysconfigdata_path = find_sysconfigdata(&cross_compile_config)?;
-    InterpreterConfig::from_sysconfigdata(&parse_sysconfigdata(sysconfigdata_path)?)
+    cross_compile_config: &CrossCompileConfig,
+) -> Result<Option<InterpreterConfig>> {
+    if let Some(path) = find_sysconfigdata(cross_compile_config)? {
+        Ok(Some(InterpreterConfig::from_sysconfigdata(
+            &parse_sysconfigdata(path)?,
+        )?))
+    } else {
+        Ok(None)
+    }
 }
 
-fn windows_hardcoded_cross_compile(
-    cross_compile_config: CrossCompileConfig,
-) -> Result<InterpreterConfig> {
-    let version = cross_compile_config.version.or_else(get_abi3_version)
-        .ok_or("PYO3_CROSS_PYTHON_VERSION or an abi3-py3* feature must be specified when cross-compiling for Windows.")?;
+/// Generates "default" cross compilation information for the target.
+///
+/// This should work for most CPython extension modules targeting
+/// Windows, MacOS and Linux.
+fn default_cross_compile(cross_compile_config: &CrossCompileConfig) -> Result<InterpreterConfig> {
+    let version = cross_compile_config
+        .version
+        .or_else(get_abi3_version)
+        .ok_or(
+            "PYO3_CROSS_PYTHON_VERSION or an abi3-py3* feature must be specified \
+            when cross-compiling and PYO3_CROSS_LIB_DIR is not set.",
+        )?;
 
     let abi3 = is_abi3();
     let implementation = PythonImplementation::CPython;
```

Comment:
While you're modifying this stuff, it might be nice to have a `PYO3_CROSS_PYTHON_IMPLEMENTATION` to choose between PyPy and CPython.

Reply:
I think it is a good idea. I'm not sure that the current PyPy implementation supports `abi3` though, which will limit its usability in many cases.

I'll add `PYO3_CROSS_PYTHON_IMPLEMENTATION` in the next revision.

Reply:
Postponed and will be addressed in a separate PR.